cornac/eval_methods/base_method.py

```diff
@@ -272,6 +272,7 @@ def __init__(
         self.val_set = None
         self.rating_threshold = rating_threshold
         self.exclude_unknowns = exclude_unknowns
+        self.mode = kwargs.get("mode", None)
```

Comment:
If `mode` is only used for next-item evaluation, I recommend to move it out of the base method

Reply:
Updated